boschshcpy/models_impl.py

```diff
@@ -117,6 +117,55 @@ def summary(self):
         super().summary()
 
 
+class SHCSmartPlugCompact(SHCDevice):
+    from .services_impl import (
+        PowerMeterService,
+        PowerSwitchProgramService,
+        PowerSwitchService,
+        CommunicationQualityService
+    )
+
+    def __init__(self, api, raw_device):
+        super().__init__(api, raw_device)
+
+        self._powerswitch_service = self.device_service("PowerSwitch")
+        self._powerswitchprogram_service = self.device_service("PowerSwitchProgram")
+        self._powermeter_service = self.device_service("PowerMeter")
+        self._communicationquality_service = self.device_service("CommunicationQuality")
+
+    @property
+    def state(self) -> PowerSwitchService.State:
+        return self._powerswitch_service.value
+
+    @state.setter
+    def state(self, state: bool):
+        self._powerswitch_service.put_state_element(
+            "switchState", "ON" if state else "OFF"
+        )
+
+    @property
+    def energyconsumption(self) -> float:
+        return self._powermeter_service.energyconsumption
+
+    @property
+    def powerconsumption(self) -> float:
+        return self._powermeter_service.powerconsumption
+
+    @property
+    def communicationquality(self) -> float:
```

Comment:
Please substitute return type `float` by  `CommunicationQualityService.State`.

Reply:
Thanks for reviewing. Just fixed it!